DEEP_DIVES/package-managers/README.md

```diff
@@ -0,0 +1,121 @@
+  12. [Possible to install and verify packages when offline](offlineavailability-install.jpg)
+  
+- **Potential benefit:** Reproducibility
+
+  13. [Easy to declare/verify the provenance of everything you run](reproducibility-provenance.jpg)
+  
+- **Potential benefits:** Forkability
+
+  14. [Users can easily fork existing registries](forkability-forkregistries.jpg)
+  15. [Users can easily start their own registries](forkability-startregistries.jpg)
+  
+- **Potential benefits:** User-hosted
+
+  16. [Users can replicate and provide packages they care about](userhosted-favoritepackages.jpg)
+  17. [Users automatically cache items that they rely on](userhosted-caching.jpg)
+  
+- **Potential benefits:** Decentralised publishing
+
+  18. [Community doesn’t need to rely on centralised registries](decentralizedpub-fewercentralregistries.jpg)
+  19. [Users can publish directly to IPFS](decentralizedpub-publishtoipfs.jpg)
+  20. [Not as many central points of failure — or of trust](decentralizedpub-fewercentralpoints.jpg)
+
+
+## Key takeaways from ranking exercise
+
+For the dot-ranking exercise, we discussed each of the 20 pain points/potential benefits individually, but focused the discussion on the items in which the dots were divergent between participants. Since our participants came from a variety of backgrounds and areas of focus -- particularly in that we had a mix of package manager users of various backgrounds and preferred languages, as well as a package manager maintainer -- the most interesting nuances were to be found in the areas where opinions differed. See the below for key takeaways.
+*Source: @achingbrain*
+
+- Connectivity is important, eg. working offline
+- Replication of content not important to end users
+- New paradigms of package management enable new use cases
+- Reproducibility! Very important
+- Decentralised publishing is a great goal
+- Decentralised registries not massively important
+- Excessive network activity is slow, user hosted packages would help
+- Why refetch assets/tarballs/etc that we know will not have changed? - e.g. redownloading is bad
+- Performance is not that important as long as it's 'fast enough'
+- Performance regressions the worst
+- Left-pad type issues are not common but are show stoppers when they do
+- Being able to reproduce builds when dependencies change over time very important
+- Cost of hosting private/internal registries not important
+- Deduping package contents important for developers but less so for maintainers, particularly if they dedupe already
+- Package creators care more about self-hosting than package users
+- All participants liked the option to fork, but not very often -- it's most useful if need to modify software for yourself
+
+## Guix patch notes that may be helpful in other package manager efforts
+*Source: @momack2*
+
+- Want executable bit
+- Create an IPLD node with that metadata using CBOR with a pointer to package and whatever key values 
```

Comment:
I can't help but think we could enable most use cases with something like this.  Call it UnixFSv1.5.  The `content` linked to could be a UnixFSv1 file/dir/hamt/whatever which gives users an upgrade path without having to re-encode all of their data.

```javascript
{
  mtime: 1561879007581
  perm: '0777',
  // ... more metadata here
  content: {
    '/': 'Qmfoo' // link to a file or dir or cbor, etc..
  }
}
```

Reply:
@achingbrain Can you please add this note to the actual md file so it’s captured inline?

Reply:
ping @achingbrain :)

Reply:
Done!